src/proof/proof_manager.cpp

```diff
@@ -561,8 +561,8 @@ void LFSCProof::toStream(std::ostream& out, const ProofLetMap& map) const
 void LFSCProof::toStream(std::ostream& out) const
 {
   TimerStat::CodeTimer proofProductionTimer(
-      *ProofManager::currentPM()->getProofProductionTime());
-
+      ProofManager::currentPM()->getStats().d_proofProductionTime);
+  ProofManager::currentPM()->getStats().d_skeletonProofTraceTime.start();
```

Comment:
Instead of manually starting/stopping timers, you can also use a `CodeTimer` within a block (just a set of curly braces). The advantage of that is that the timer automatically gets stopped, no matter how the block is exited (e.g. due to an exception).

Reply:
That's neat -- I didn't know about the exception interaction.

Sounds good. One of these days I should split this method up too.